lib/determine-basal/cob.js

```diff
@@ -66,11 +65,18 @@ function detectCarbAbsorption(inputs) {
                 continue;
             }
         }
+        if (bucketed_data[bucketed_data.length-1].display_time) {
+            lastbgTime = new Date(bucketed_data[bucketed_data.length-1].display_time.replace('T', ' '));
+        } else if ((lastbgi >= 0) && glucose_data[lastbgi].display_time) {
+            lastbgTime = new Date(glucose_data[lastbgi].display_time.replace('T', ' '));
+        } else if ((lastbgi >= 0) && glucose_data[lastbgi].dateString) {
+            lastbgTime = new Date(glucose_data[lastbgi].dateString);
+        } else { console.error("Could not determine last BG time"); }
```

Comment:
Why did you move these?

Reply:
Never mind, I see the commit comment now: “Move lastbgTime determination past code that skips old data to avoid unmeaningful errors in the log.”